drivers/disk/nvme/nvme_disk.c

```diff
@@ -194,6 +193,11 @@ static int nvme_disk_ioctl(struct disk_info *disk, uint8_t cmd, void *buff)
 	return ret;
 }
 
+static int nvme_disk_init(struct disk_info *disk)
+{
+	return nvme_disk_ioctl(disk, DISK_IOCTL_CTRL_INIT, NULL);
+}
+
```

Comment:
Since we end up calling `ioctl` with `DISK_IOCTL_CTRL_INIT`, should we consider dropping the `init` callback and replace contents of `disk_access_init` with call to the IOCTL, seems that they have the same code and we can save on some callbacks.

Reply:
I thought about this- the reason I don't want to drop `init` (yet) is because this would be a stable API change in that case. The documentation added in this PR already guides users to use the IOCTLs over `disk_access_init`, in the future we can (and should) deprecate `disk_access_init`. I just want to avoid doing it now to give users a transition window.

Do you think it would be best to mark `disk_access_init` as deprecated in this PR, and start the stable API change process? We can then remove `disk_access_init` in the next release cycle

Reply:
I do not mean in this pr, I mean should we log issue planning to do that.